lighthouse-core/audits/byte-efficiency/unminified-javascript.js

```diff
@@ -77,21 +77,24 @@ class UnminifiedJavaScript extends ByteEfficiencyAudit {
     /** @type {Array<LH.Audit.ByteEfficiencyItem>} */
     const items = [];
     const warnings = [];
-    for (const requestId of Object.keys(artifacts.Scripts)) {
-      const scriptContent = artifacts.Scripts[requestId];
+    for (const {requestId, code} of artifacts.Scripts) {
+      if (!code) continue;
       const networkRecord = networkRecords.find(record => record.requestId === requestId);
-      if (!networkRecord || !scriptContent) continue;
 
       try {
-        const result = UnminifiedJavaScript.computeWaste(scriptContent, networkRecord);
+        const result = UnminifiedJavaScript.computeWaste(code, networkRecord);
         // If the ratio is minimal, the file is likely already minified, so ignore it.
         // If the total number of bytes to be saved is quite small, it's also safe to ignore.
         if (result.wastedPercent < IGNORE_THRESHOLD_IN_PERCENT ||
           result.wastedBytes < IGNORE_THRESHOLD_IN_BYTES ||
           !Number.isFinite(result.wastedBytes)) continue;
         items.push(result);
       } catch (err) {
-        warnings.push(`Unable to process ${networkRecord.url}: ${err.message}`);
+        if (networkRecord) {
```

Comment:
can we have a fallback like for `url` in `computeWaste()` since we know it's only scripts in the main page missing the network record?